strawberry/contrib/django/type.py

```diff
@@ -0,0 +1,74 @@
+import strawberry
+from strawberry.contrib.django.converter import convert_django_field
+from strawberry.type import type as strawberry_type
+
+
+def model_type(
```

Comment:
I wondering if we can find a nicer name for this? :D

Also, have you considered doing a django approach, like

```
class ABC(ModelType):
    class Meta:
        ...
```

to be consistent with django's approach? I prefer decorators but it could be worth thinking about this :) 

Pinging @adamchainz if he has some suggestion (and time to look at this) ;)

Reply:
A meta class might work but I'm not sure about the strawberry internals so I can't really advise on it. I guess it'd be not too much work to convert this class decorator since you just need to reorder it so strawberry's `ModelType` executes the current code at the end of class definition time (on python 3.6+ you can use `__init_subclass__` instead of a metaclass).

Reply:
Sure! Any ideas? Maybe `django_object_type` to keep it consistent with Graphene lingo, or `django_model_type` or `django_type`?

I had thought of trying to use a meta class early on, but decided to go with a decorator to keep things consistent with the rest of the library. I also wanted to be able to tap into the power of strawberry's existing `@strawberry.type` decorator, and this seemed like the simplest way to do that. 

I'm happy to experiment with metaclasses if you think it would be beneficial to have them though :)

Reply:
don't know, my thinking was around consistency with what django users are familiar with, but again, this is something that can be added later :)

Let's also stick with model_type for now, should be easy to change later if needed :)

Reply:
@patrick91 @adamchainz After thinking this over and experimenting with both approaches, I think we should stick with a decorator. Using a decorator maximizes compatibility with the rest of the library and keeps our developer experience consistent.

However, I think we should try to keep the developer experience consistent with Django's patterns wherever we can as well. To accomplish this, I've updated the `model_type` decorator to allow developers to use either decorator arguments or `Meta` class attributes to configure a model type. Developers can now define model types using either of these syntaxes:

```python
@model_type(model=Todo, fields=['id', 'name', 'done'])
class TodoType:
    pass

# OR

@model_type
class TodoType:
    class Meta:
        model = Todo
        fields = ['id', 'name', 'done']
```

This seems like a happy medium to me. Would love your thoughts on this approach :)